### PR TITLE
shutdown handler should be a singleton.

### PR DESCRIPTION
**Pull request by heavenlyhash**
Use of multiple handlers as written will result in the first one of them to finish exiting the whole process without waiting for other handler instances, which may result in incomplete cleanup.  There is now only one allowed handler instance.

### CODE REVIEW

**Comment by lmars** [comment]
LGTM

**Comment by archSeer** [comment]
LGTM

File: host/host.go [hunk 1]
```diff
@@ -298,9 +297,9 @@ func runDaemon(args *docopt.Args) {
 		g.Log(grohl.Data{"at": "sampi_disconnected", "err": jobStream.Err})
 
 		// if the process is shutting down, just block
-		if sh.Active {
-			<-make(chan struct{})
-		}
+		//if shutdown.Active { // Wot?
```
Comment (heavenlyhash) [inline host/host.go]: Sanity check, can we remove this?  I'm almost certain the other loop behavior continuing on here is fine.

Comment (heavenlyhash) [inline host/host.go]: Belay that, by the light of the morning I understand what this is for.

File: pkg/shutdown/handler.go [hunk 2]
```diff
@@ -7,33 +7,41 @@ import (
...
 	"syscall"
 )
 
-type Handler struct {
-	Active bool
+var h = newHandler()
 
-	mtx  sync.RWMutex
-	done chan struct{}
+type Handler struct {
```
Comment (titanous) [inline pkg/shutdown/handler.go]: I think we can unexport the `Handler` type?

**Pull request merged**